src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java

```diff
@@ -206,7 +207,13 @@ public void onServiceLost(NsdServiceInfo service) {
 
   @Override
   public void onDiscoveryStopped(String serviceType) {
-    Log.i(TAG, "Discovery stopped: " + serviceType);
+    Log.i(
+        TAG,
+        "Discovery stopped: " + serviceType + ", discoveredServiceInfo: " + discoveredServiceInfo);
+    if (discoveredServiceInfo != null) {
+      resolveService(discoveredServiceInfo, callbackHandle, contextHandle, chipMdnsCallback);
```

Comment:
this changes behavior: previously we would call this as soon as we would receive the entry, now we only call this on stopped. What is the lifetime here? I would generally expect long(ish) running resolve followed by a stop, so if that is the case this introduces significant latency.

Can we keep the resolve call inside the onServiceFound and just call on the first value received? or is it mandatory to have the last value found reported?

Reply:
1. As soon as the service is found using discoverService, the stop command is called with the code below, and when the stop is completed, the corresponding method is called.
https://github.com/project-chip/connectedhomeip/blob/7a54490fc3e37a67692073922ec7ee6f15c48473/src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java#L101

-> I think resolveService works when discoverService is completely terminated I decided that it was right to do so, so I modified it as follows. Do you think it would be more appropriate to call it directly?

2. The modified code also calls only by utilizing the first Service found.

Reply:
@andy31415 please check my opinion.

Reply:
@andy31415 Please check about this.

Reply:
Ok. A comment would be useful explaining that only a single result is ever returned (which I am unclear if it is ideal: dnssd may be split under multiple packets and you may miss some IP addresses).

As long as this passes your tests, we can merge.